content/docs/1.5.1/volumes-and-nodes/internal-data-structures.md

```diff
@@ -0,0 +1,124 @@
+
+```
+revision.counter
+volume-head-000.img
+volume-head-000.img.meta
+volume.meta
+```
+
+- revision.counter: Records the latest revision count of the replica for the volume.
+- volume-head-000.img: The actual data file containing the volume data.
+- volume-head-000.img.meta: Stores metadata related to the volume-head-000.img file, including the file name, parent file, creation timestamp, and other attributes.
+- volume.meta: Records the attributes of the volume, such as size, current head file, parent snapshot, and other metadata.
+
+After creating multiple snapshots, the disk chain will look like this:
+
+```
+snap 1 -----> snap 2 -----> ... -----> snap N -----> volume head
+```
+
+The file list of a practical volume containing two snapshots and one volume head might look like this:
+
+```
+revision.counter
+volume-head-002.img
+volume-head-002.img.meta
+volume-snap-2e8bf0e9-3751-479c-a560-96a275d2672c.img
+volume-snap-2e8bf0e9-3751-479c-a560-96a275d2672c.img.meta
+volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img
+volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img.meta
+volume.meta
+```
+
+In this example, the disk chain is:
+```
+96a275d2672c -----> 9a61c5ff0907 -----> volume-head-002
+```
+Where 96a275d2672c and 9a61c5ff0907 are abbreviations for the snapshot file names.
+
+The contents of the volume.meta and image metadata files (*.img.meta) are:
+```
+// volume.meta
+{
+  "Size": 2147483648,
+  "Head": "volume-head-002.img",
+  "Dirty": true,
+  "Rebuilding": false,
+  "Error": "",
+  "Parent": "volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img",
+  "SectorSize": 512,
+  "BackingFilePath": ""
+}
+
+// volume-head-002.img.meta
+{
+  "Name": "volume-head-002.img",
+  "Parent": "volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img",
+  "Removed": false,
+  "UserCreated": false,
+  "Created": "2024-03-09T07:33:16Z",
+  "Labels": null
+}
+
+// volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img.meta
+{
+  "Name": "volume-head-001.img",
+  "Parent": "volume-snap-2e8bf0e9-3751-479c-a560-96a275d2672c.img",
+  "Removed": false,
+  "UserCreated": true,
+  "Created": "2024-03-09T07:33:16Z",
+  "Labels": null
+}
+
+// volume-snap-2e8bf0e9-3751-479c-a560-96a275d2672c.img.meta
+{
+  "Name": "volume-head-000.img",
+  "Parent": "",
+  "Removed": false,
+  "UserCreated": true,
+  "Created": "2024-03-09T07:32:51Z",
+  "Labels": null
+}
+```
+
+The `.img.meta` files store metadata related to the corresponding `.img` data files in the disk chain. The metadata files contain information about the relationships between the data files, which helps maintain the integrity of the disk chain and facilitate operations such as snapshots and rebuilds.
+
+> **NOTICE**
+>
+> You can ignore the Name field in the .img.meta. The Parent field refers to the name of the parent .img file.
```

Comment:
Does this mean the name filed is invalid?

Reply:
Yes, it is unused.